Sourcery/Models/TypeName.swift

```diff
@@ -204,6 +205,37 @@ public final class TypeName: NSObject, AutoCoding, AutoEquatable, AutoDiffable,
 
 }
 
+/// Descibes Swift class
+public final class GenericType: NSObject, SourceryModel {
+    /// the name of the base type.
+    ///
+    /// `Array<Int>`'s GenericType.name is `Array`
+    public let name: String
+    
+    public let referencedTypes: [Type]
```

Comment:
I'll add them now :) Still trying to figure out some edge-case scenarios, definitely didn't think about the two annotations.